pennylane/devices/default_mixed.py

```diff
@@ -0,0 +1,145 @@
+:mod:`qubit operations <pennylane.ops.qubit>`, providing a simple mixed-state simulation of
+qubit-based quantum circuits.
+"""
+
+import numpy as np
+
+from pennylane import QubitDevice
+from pennylane.operation import DiagonalOperation, Channel
+
+
+class DefaultMixed(QubitDevice):
+    """Default qubit device for performing mixed-state computations in PennyLane.
+
+    Args:
+        wires (int, Iterable[Number, str]): Number of subsystems represented by the device,
+            or iterable that contains unique labels for the subsystems as numbers
+            (i.e., ``[-1, 0, 2]``) or strings (``['ancilla', 'q1', 'q2']``).
+        shots (int): Number of times the circuit should be evaluated (or sampled) to estimate
+            the expectation values. Defaults to 1000 if not specified.
+            If ``analytic == True``, the number of shots is ignored
+            in the calculation of expectation values and variances, and only controls the number
+            of samples returned by ``sample``.
+        analytic (bool): indicates if the device should calculate expectations
+            and variances analytically.
+    """
+
+    name = "Default mixed-state qubit PennyLane plugin"
+    short_name = "default.mixed"
+
+    operations = {
+        "BasisState",
+        "QubitStateVector",
+        "QubitUnitary",
+        "DiagonalQubitUnitary",
+        "PauliX",
+        "PauliY",
+        "PauliZ",
+        "MultiRZ",
+        "Hadamard",
+        "S",
+        "T",
+        "CNOT",
+        "SWAP",
+        "CSWAP",
+        "Toffoli",
+        "CZ",
+        "PhaseShift",
+        "RX",
+        "RY",
+        "RZ",
+        "Rot",
+        "CRX",
+        "CRY",
+        "CRZ",
+        "CRot",
+        "AmplitudeDamping",
+        "GeneralizedAmplitudeDamping",
+        "PhaseDamping",
+        "DepolarizingChannel",
+        "QubitChannel",
+    }
+
+    def __init__(self, wires, *, shots=1000, analytic=True):
+        # call QubitDevice init
+        super().__init__(wires, shots, analytic)
+
+        # Create the initial state.
+        self._state = self._create_basis_state(0)
+        self._pre_rotated_state = self._state
+
+    def _create_basis_state(self, index):
+        """Return the density matrix representing a computational basis state over all wires.
+
+        Args:
+            index (int): integer representing the computational basis state.
+
+        Returns:
+            array[complex]: complex array of shape ``[2] * (2 * num_wires)``
+            representing the density matrix of the basis state.
+        """
+        rho = np.zeros((2 ** self.num_wires, 2 ** self.num_wires), dtype=np.complex128)
+        rho[index, index] = 1
+        rho = self._asarray(rho, dtype=self.C_DTYPE)
+        return self._reshape(rho, [2] * (2 * self.num_wires))
+
+    @property
+    def state(self):
+        """Returns the state density matrix of the circuit prior to measurement"""
```

Comment:
@trbromley - perhaps this sets the precedent for `qml.state()` returning whatever the _device_ claims the state to be? 🙂 So in this case, it would return a density matrix

Reply:
This might work now, as we could take advantage of the `output_dim` 'correction' in the quantum tape.

Reply:
Nice! Although at some point I think we should think about this terminology. I could imagine the pure state simulators having a `density_matrix` property as well, e.g., used when returning the state on subsystems.

Reply:
🤔